addon/mixins/sortable-item.js

```diff
@@ -308,14 +308,18 @@ export default Mixin.create({
     const groupDirection = this.get('group.direction');
     let dragOrigin;
     let elementOrigin;
+    let scrollOrigin;
 
     if (groupDirection === 'x') {
       dragOrigin = getX(startEvent);
       elementOrigin = this.get('x');
+      scrollOrigin = $(this.element).offset().left;
 
       return event => {
         let dx = getX(event) - dragOrigin;
-        let x = elementOrigin + dx;
+        let elementX = this.get('x') - this.element.offsetLeft;
+        let scrollX = $(this.element).offset().left - elementX;
+        let x = elementOrigin + dx + (scrollOrigin - scrollX);
```

Comment:
Here we basically account for the scroll to determine the real `dx`, instead of just relying on the distance from the mouse starting point (which will not change if you scroll an element that is parent of the list). So the actual `dx` is the mouse offset + the difference in scroll from when you started dragging.

Once again, not entirely sure this is the best approach!